pages/user/[username]/index.tsx

```diff
@@ -106,38 +106,25 @@ Contributor.PageLayout = ProfileLayout;
 export default Contributor;
 
 export const getServerSideProps = async (context: UserSSRPropsContext) => {
-  return await handleUserSSR(context);
-};
-
-export type UserSSRPropsContext = GetServerSidePropsContext<{ username: string }>;
-
-export async function handleUserSSR({ params }: GetServerSidePropsContext<{ username: string }>) {
-  const { username } = params!;
-
-  async function fetchUserData() {
-    try {
-      const req = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${username}`, {
-        headers: {
-          accept: "application/json",
-        },
-      });
-
-      if (!req.ok) {
-        return null;
-      }
+  const { username } = context.params!;
+  const req = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${username}`, {
+    headers: {
+      accept: "application/json",
+    },
+  });
 
-      return (await req.json()) as DbUser;
-    } catch (e) {
-      // eslint-disable-next-line no-console
-      console.error(e);
-      return null;
-    }
+  if (!req.ok) {
+    return {
+      redirect: { destination: "/404" },
```

Comment:
Next has a built-in property we can return for a 404

```suggestion
      notFound: true,
```